arangod/RocksDBEngine/RocksDBOptionFeature.cpp

```diff
@@ -931,10 +1234,26 @@ rocksdb::BlockBasedTableOptions RocksDBOptionFeature::doGetTableOptions()
 
   result.block_size = _tableBlockSize;
   result.filter_policy.reset(rocksdb::NewBloomFilterPolicy(10, true));
-  // use slightly space-optimized format version 3
-  result.format_version = 3;
+  result.enable_index_compression = _enableIndexCompression;
+  result.format_version = _formatVersion;
+  result.prepopulate_block_cache =
+      _prepopulateBlockCache
+          ? rocksdb::BlockBasedTableOptions::PrepopulateBlockCache::kFlushOnly
+          : rocksdb::BlockBasedTableOptions::PrepopulateBlockCache::kDisable;
+  result.reserve_table_builder_memory = _reserveTableBuilderMemory;
+  result.reserve_table_reader_memory = _reserveTableReaderMemory;
   result.block_align = _blockAlignDataBlocks;
 
+  if (_checksumType == "xxHash") {
+    result.checksum = rocksdb::ChecksumType::kxxHash;
+  } else if (_checksumType == "xxHash64") {
+    result.checksum = rocksdb::ChecksumType::kxxHash64;
+  } else if (_checksumType == "XXH3") {
+    result.checksum = rocksdb::ChecksumType::kXXH3;
+  } else {
+    result.checksum = rocksdb::ChecksumType::kCRC32c;
+  }
```

Comment:
Should we log a warning if `_checksum != "crc32"`?

Reply:
Done